core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -158,18 +171,33 @@ private TopNOperator(
         return pages;
     }
 
+    private boolean isBuilderFull()
+    {
+        return maxPartialMemory <= topNProcessor.getEstimatedSizeInBytes();
```

Comment:
nit: it's more natural to use `topNProcessor.getEstimatedSizeInBytes() >= maxPartialMemory`

Reply:
Done